run.py

```diff
@@ -11,6 +11,28 @@
 from mig_dx_api._dataset import DatasetOperations
 from google.cloud import bigquery
 
+def get_target_installation(installations: list[Installation], target_installation_id: str) -> Installation:
+    if len(installations) == 0:
+        raise Exception("No valid installations found")
+    elif len(installations) == 1:
+        target_install = installations[0]
+        # error if the target installation id doesn't match the only existing installation
+        if target_installation_id:
+            if target_install.movement_app_installation_id != int(target_installation_id):
```

Comment:
I'd combine this conditional with the one on line 20 with an `and`. I don't see a compelling reason for them to be on separate lines.